ttnn/CMakeLists.txt

```diff
@@ -657,7 +697,92 @@ if(WITH_PYTHON_BINDINGS)
     )
 endif()
 
-add_library(ttnn SHARED ${TTNN_FINAL_SRC})
+function(use_precompiled_header TARGET)
+    if("${PRECOMPILED_HEADER_TARGET}" STREQUAL "")
+        set(PRECOMPILED_HEADER_TARGET "${TARGET}" PARENT_SCOPE)
+        target_precompile_headers(${TARGET} PUBLIC ${TTNN_PRECOMPILED_HEADERS})
+    else()
+        target_precompile_headers(${TARGET} REUSE_FROM ${PRECOMPILED_HEADER_TARGET})
+    endif()
+endfunction()
+
+set(PRECOMPILED_HEADER_TARGET "")
+
+function(add_ttnn_sublibrary SUBLIBRARY_NAME)
+    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
does this also cover RelWithDebInfo?

Reply:
It shouldn't, because that would be a different build type.